components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/model/template/service/BallerinaPath.java

```diff
@@ -96,4 +106,18 @@ public BallerinaPath getDefaultValue() {
     public Set<Map.Entry<String, BallerinaOperation>> getOperations() {
         return operations;
     }
+
+
+    /**
+     * Method to set the Path level server opeartions to operation level servers,
+     * if operation level servers are not present
+     *
+     * @param operation {@link Operation} The current considered operation object
+     * @param pathLevelServers list if server objects defined in the open API PAth level
+     */
+    private void setServersToOperationLevel(Operation operation, List<Server> pathLevelServers) {
+        if (operation.getServers() == null && pathLevelServers != null) {
```

Comment:
operation.getExtensions also needs to be considered here

Reply:
Yes, since we do not support adding path level endpoints, we can add this as an improvement. But in the open API spec servers can be added at path level, so we have to support it. We can add adding extensions also at path level as an improvement. Please create an issue for this.

Reply:
Actually my point here is different. It is the logic. If there is an operation with custom extension along with a path level server object, this logic will assign the path level object rather than the value in the custom extension.

Reply:
The priority alway given to the extension, If the extension is not present then only the servers object is look at. So if there is an operation level extension ,that value will get chosen, not the path level server value